pylint/config/arguments_manager.py

```diff
@@ -161,3 +225,274 @@ def _parse_command_line_configuration(
+        """Write a configuration file according to the current configuration
+        into the given stream or stdout
+        """
+        options_by_section: Dict[str, List[Tuple]] = {}
+        sections = []
+        for provider in self.options_providers:
+            for section, options in provider.options_by_section():
+                if section is None:
+                    section = provider.name
+                if section in skipsections:
+                    continue
+                options = [
+                    (n, d, v)
+                    for (n, d, v) in options
+                    if d.get("type") is not None and not d.get("deprecated")
+                ]
+                if not options:
+                    continue
+                if section not in sections:
+                    sections.append(section)
+                all_options = options_by_section.setdefault(section, [])
+                all_options += options
+        stream = stream or sys.stdout
+        printed = False
+        for section in sections:
+            if printed:
+                print("\n", file=stream)
+            utils.format_section(
+                stream, section.upper(), sorted(options_by_section[section])
+            )
+            printed = True
+
+    def load_provider_defaults(self):
+        """Initialize configuration using default values."""
+        for provider in self.options_providers:
+            provider.load_defaults()
+
+    def read_config_file(
+        self, config_file: Optional[Path] = None, verbose: bool = False
+    ) -> None:
+        """Read the configuration file but do not load it (i.e. dispatching
+        values to each option's provider)
+        """
+        if config_file:
+            config_file = Path(os.path.expandvars(config_file)).expanduser()
+            if not config_file.exists():
+                raise OSError(f"The config file {str(config_file)} doesn't exist!")
+
+            parser = self.cfgfile_parser
+            if config_file.suffix == ".toml":
+                try:
+                    self._parse_toml(config_file, parser)
+                except tomllib.TOMLDecodeError:
+                    pass
+            else:
+                # Use this encoding in order to strip the BOM marker, if any.
+                with open(config_file, encoding="utf_8_sig") as fp:
+                    parser.read_file(fp)
+                # normalize each section's title
+                for sect, values in list(parser._sections.items()):
+                    if sect.startswith("pylint."):
+                        sect = sect[len("pylint.") :]
+                    if not sect.isupper() and values:
+                        parser._sections[sect.upper()] = values
+
+        if not verbose:
+            return
+        if config_file and config_file.exists():
+            msg = f"Using config file '{config_file}'"
+        else:
+            msg = "No config file found, using default configuration"
```

Comment:
This will never be executed because of line 396.

Reply:
I think it can be executed if L393 fails?

Reply:
You're right.